Rename menu components (#2996)
vivid-planet/comet: docs/docs/migration/migration-from-v7-to-v8.md

    }
    ```

> ### Rename `Menu` and related components to `MainNavigation` in `@comet/admin`

Reviewer: Should the code-migration script be mentioned here? 🤔 

https://github.com/vivid-planet/comet-upgrade/pull/45

Author: Good point - thank you! Added here: 63a4d0763cd94a742525326d72bb403a954f3c50

Reviewer: Ah, sorry, I thought these upgrade-scripts had to be run manually and that's why I thought the cli-command should be mentioned in the docs. 

This isn't the case however, as (I think) your upgrade-script would be run automatically when upgrading to `v8` with `npx @comet/upgrade@latest v8`. So this mention in the docs isn't actually necessary.

> To better differentiate between imports from `@comet/admin` and `mui`, `Menu` and its related components are renamed:

Reviewer: Should this just be the same as in the changeset? 

```suggestion
To better differentiate between imports from `@comet/admin` and `@mui/material`, the following components and related types have been renamed:
```

-   `Menu` → `MainNavigation`
-   `MenuProps` → `MainNavigationProps`
-   `MenuClassKey` → `MainNavigationClassKey`
-   `MenuItem` → `MainNavigationItem`
-   `MenuItemProps` → `MainNavigationItemProps`
-   `MenuItemClassKey` → `MainNavigationItemClassKey`
-   `MenuCollapsibleItem` → `MainNavigationCollapsibleItem`
-   `MenuCollapsibleItemProps` → `MainNavigationCollapsibleItemProps`
-   `MenuCollapsibleItemClassKey` → `MainNavigationCollapsibleItemClassKey`

> -   `IMenuContent` → `IMainNavigation`
> -   `IWithMenu` → `IWithMainNavigation`

Reviewer: ```suggestion
-   `IMenuContext` → `MainNavigationContext`
-   `IWithMenu` → `WithMainNavigation`
```
